### PR TITLE
feat: Permissions API

### PR DESCRIPTION
**Pull request by mlynch**
This adds a new API called `Permissions` that lets a developer check if certain permissions have been granted yet before requesting them.

It largely follows the [Permissions API](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API) in the browser.

Examples:

```typescript
  async checkCamera() {
    const has = await Plugins.Permissions.query({ name: PermissionType.Camera });
  }

  async checkPhotos() {
    const has = await Plugins.Permissions.query({ name: PermissionType.Photos });
  }

  async checkGeo() {
    const has = await Plugins.Permissions.query({ name: PermissionType.Geolocation });
  }

  async checkPush() {
    const has = await Plugins.Permissions.query({ name: PermissionType.PushNotifications });
  }
```

Also couldn't help myself and fixed bug in the Geolocation API for Android where calling `requestPermissions` would cause the `requestPermissions` call to return over and over again, which really should only happen when the requestPermission called happened on `watchPosition`. I don't think this method is really used anyways

Also fixed a bug in Camera where it was creating the image picker on a background thread

### CODE REVIEW

File: site/docs-md/apis/permissions/index.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+title: Permissions
+description: Permissions API
+url: /docs/apis/permissions
+contributors:
+  - mlynch
+---
+
+<plugin-platforms platforms="pwa,electron,ios,android"></plugin-platforms>
+
+# Photos
```
Comment (jcesarmobile) [inline site/docs-md/apis/permissions/index.md]:
```suggestion
# Permissions
```

File: ios/Capacitor/Capacitor/Plugins/Permissions.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import Foundation
+import Photos
+
+/**
+ * Implement three common modal types: alert, confirm, and prompt
```
Comment (jcesarmobile) [inline ios/Capacitor/Capacitor/Plugins/Permissions.swift]:
```suggestion
 * Allow to check hardware permissions
```

File: example/src/pages/camera/camera.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Component, NgZone } from '@angular/core';
 import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
-import { IonicPage, NavController, NavParams } from 'ionic-angular';
+import { IonicPage, NavController, NavParams, AlertController } from 'ionic-angular';
```
Comment (jcesarmobile) [inline example/src/pages/camera/camera.ts]:
```suggestion
import { IonicPage, NavController, NavParams } from 'ionic-angular';
```
remove unused `AlertController`

File: example/src/pages/camera/camera.ts [hunk 4]
```diff
@@ -25,14 +25,17 @@ const { Filesystem } = Plugins;
 export class CameraPage {
   image: SafeResourceUrl;
 
-  constructor(public navCtrl: NavController, public navParams: NavParams, private zone: NgZone, private sanitizer: DomSanitizer) {
+  constructor(public navCtrl: NavController,
```
Comment (jcesarmobile) [inline example/src/pages/camera/camera.ts]:
```suggestion
  constructor(public navCtrl: NavController, public navParams: NavParams, private zone: NgZone, private sanitizer: DomSanitizer) {
```

File: example/src/pages/camera/camera.ts [hunk 5]
```diff
@@ -25,14 +25,17 @@ const { Filesystem } = Plugins;
...
+              public navParams: NavParams,
```
Comment (jcesarmobile) [inline example/src/pages/camera/camera.ts]:
```suggestion
           
```

File: example/src/pages/camera/camera.ts [hunk 6]
```diff
@@ -25,14 +25,17 @@ const { Filesystem } = Plugins;
...
+              private zone: NgZone,
```
Comment (jcesarmobile) [inline example/src/pages/camera/camera.ts]:
```suggestion
            
```

File: example/src/pages/camera/camera.ts [hunk 7]
```diff
@@ -25,14 +25,17 @@ const { Filesystem } = Plugins;
...
+              private sanitizer: DomSanitizer,
```
Comment (jcesarmobile) [inline example/src/pages/camera/camera.ts]:
```suggestion
            
```

File: example/src/pages/camera/camera.ts [hunk 8]
```diff
@@ -25,14 +25,17 @@ const { Filesystem } = Plugins;
...
+              private alertCtrl: AlertController) {
```
Comment (jcesarmobile) [inline example/src/pages/camera/camera.ts]:
```suggestion
            
```

**Comment by SynovialWheat9011** [comment]
Ok.

**Comment by SynovialWheat9011** [comment]
Hi ! Make a way to navigate with Ionic.

**Pull request merged**